scripts/btsbot/btsbot.py

```diff
@@ -0,0 +1,255 @@
+      title={The Zwicky Transient Facility Bright Transient Survey. III. $\texttt{BTSbot}$: Automated Identification and Follow-up of Bright Transients with Deep Learning}, 
+      author={Nabeel Rehemtulla and Adam A. Miller and Theophile Jegou Du Laz and Michael W. Coughlin and Christoffer Fremling and Daniel A. Perley and Yu-Jing Qin and Jesper Sollerman and Ashish A. Mahabal and Russ R. Laher and Reed Riddle and Ben Rusholme and Shrinivas R. Kulkarni},
+      year={2024},
+      eprint={2401.15167},
+      archivePrefix={arXiv},
+      primaryClass={astro-ph.IM}
+}
+"""
+
+_DESCRIPTION = """\
+This is the production version of the BTSbot training set, limited to public (programid=1) ZTF alerts.
+Article: https://arxiv.org/abs/2401.15167
+Attribution: Nabeel Rehemtulla, Adam A. Miller, Theophile Jegou Du Laz, Michael W. Coughlin, Christoffer Fremling, Daniel A. Perley, Yu-Jing Qin, Jesper Sollerman, Ashish A. Mahabal, Russ R. Laher, Reed Riddle, Ben Rusholme, Shrinivas R. Kulkarni
+"""
+
+_HOMEPAGE = "https://zenodo.org/records/10839691"
+
+_LICENSE = "CC BY 4.0"
+
+_VERSION = "10.0.0"
+
+_FLOAT_FEATURES = [
+    'jd',
+    'diffmaglim',
+    'ra',
+    'dec',
+    'magpsf',
+    'sigmapsf',
+    'chipsf',
+    'magap',
+    'sigmagap',
+    'distnr',
+    'magnr',
+    'chinr',
+    'sharpnr',
+    'sky',
+    'magdiff',
+    'fwhm',
+    'classtar',
+    'mindtoedge',
+    'seeratio',
+    'magapbig',
+    'sigmagapbig',
+    'sgmag1',
+    'srmag1',
+    'simag1',
+    'szmag1',
+    'sgscore1',
+    'distpsnr1',
+    'jdstarthist',
+    'scorr',
+    'sgmag2',
+    'srmag2',
+    'simag2',
+    'szmag2',
+    'sgscore2',
+    'distpsnr2',
+    'sgmag3',
+    'srmag3',
+    'simag3',
+    'szmag3',
+    'sgscore3',
+    'distpsnr3',
+    'jdstartref',
+    'dsnrms',
+    'ssnrms',
+    'magzpsci',
+    'magzpsciunc',
+    'magzpscirms',
+    'clrcoeff',
+    'clrcounc',
+    'neargaia',
+    'neargaiabright',
+    'maggaia',
+    'maggaiabright',
+    'exptime',
+    'drb',
+    'acai_h',
+    'acai_v',
+    'acai_o',
+    'acai_n',
+    'acai_b',
+    'new_drb',
+    'peakmag',
+    'maxmag',
+    'peakmag_so_far',
+    'maxmag_so_far',
+    'age',
+    'days_since_peak',
+    'days_to_peak',
+]
+
+_INT_FEATURES = [
+    'label',
+    'fid',
+    'programid',
+    'object_id',
+    'field',
+    'nneg',
+    'nbad',
+    'ndethist',
+    'ncovhist',
+    'nmtchps',
+    'nnotdet',
+    'N',
+    'healpix',
+]
+
+_BOOL_FEATURES = [
+    'isdiffpos',
+    'is_SN',
+    'near_threshold',
+    'is_rise',
+]
+
+_STRING_FEATURES = [
+    'OBJECT_ID_',
+    'source_set',
+    'split',
+]
+
+
+class BTSbot(datasets.GeneratorBasedBuilder):
+    """BTSbot training set"""
+
+    VERSION = _VERSION
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="BTSbot_training_set", 
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['./data/healpix=*/*.hdf5']}),
```

Comment:
This is where data splits are defined @mattgrayling.

Reply:
@mattgrayling I have modified this part of `btsbot.py` to support train/val/test splits. It now expects 3 separate files per healpix directory, with `train_001-of-001.hdf5` storing training examples only, `val_001-of-001.hdf5` storing val examples only, and `test_001-of-001.hdf5` storing test examples only.
NOTE: the lines I've added are currently commented out to avoid breaking things -- they need to be uncommented (and old line deleted) once saving has been made consistent with this format.

Reply:
Oh and if there are no train/val/test examples for a given healpix, then it's fine to omit the relevant files.